### PR TITLE
feat(object store): support azblob workload identity

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
as title, currently we use my personal fork commit for debugging, will merge this pr after [reqsign](https://github.com/Xuanwo/reqsign) release a new version.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/object_store/Cargo.toml [hunk 1]
```diff
@@ -37,6 +37,10 @@ thiserror-ext = { workspace = true }
 tokio = { version = "0.2", package = "madsim-tokio", features = ["fs"] }
 tokio-retry = "0.3"
 tracing = "0.1"
+
+[patch.crates-io]
+reqsign = { git = "https://github.com/wcy-fdu/reqsign.git", rev = "f6ff677" }
```
Comment (hzxa21) [inline src/object_store/Cargo.toml:42]: Why is the `rev` different from the one specified in Cargo.toml?

File: src/object_store/src/object/opendal_engine/azblob.rs [hunk 2]
```diff
@@ -28,14 +28,14 @@ impl OpendalObjectStore {
...
+        // let account_name = std::env::var("AZBLOB_ACCOUNT_NAME")
+        //     .unwrap_or_else(|_| panic!("AZBLOB_ACCOUNT_NAME not found from environment variables"));
+        // let account_key = std::env::var("AZBLOB_ACCOUNT_KEY")
+        //     .unwrap_or_else(|_| panic!("AZBLOB_ACCOUNT_KEY not found from environment variables"));
 
         builder.endpoint(&endpoint);
-        builder.account_name(&account_name);
-        builder.account_key(&account_key);
+        // builder.account_name(&account_name);
+        // builder.account_key(&account_key);
```
Comment (hzxa21) [inline src/object_store/src/object/opendal_engine/azblob.rs:38]: Should we just remove the commented codes?

File: src/object_store/Cargo.toml (hunk 1 above)
Comment (wcy-fdu) [reply src/object_store/Cargo.toml:42]: discussed offline.

File: risedev.yml [hunk 3]
```diff
@@ -343,6 +343,18 @@ profile:
       - use: compute-node
       - use: frontend
 
+  meta-1cn-1fe-sqlite:
```
Comment (wcy-fdu) [inline risedev.yml:346]: This change seem coming from other unrelated pr, I did not change `risedev.yml` and it's the same as main branch.

Comment (hzxa21) [reply risedev.yml:346]: Can you rebase to eliminate them from your PR?

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

**Pull request merged**